Add Cursor style

jamesnw opened the pull request.

ddbeck commented on features/cursor-style.yml.dist:
```diff
@@ -0,0 +1,126 @@
+    edge: "83"
+    firefox: ≤72
+    firefox_android: "95"
+    safari: ≤13.1
+    safari_ios: ≤13.4
```
Across the board, caniuse says that Safari doesn't support this feature but BCD says it does. See https://github.com/Fyrd/caniuse/issues/7066. I've filed https://github.com/mdn/browser-compat-data/issues/24142 to see if that can/should be fixed. I think this PR is going to be blocked for a little bit until we've figured who's correct.

(I'll wait for triage on BCD to see if it's correctable soon—if not, we can turn this into a draft feature so the PR doesn't have to be held open for forever.)

ddbeck commented on features/cursor-style.yml.dist:
```diff
@@ -0,0 +1,126 @@
+  #   firefox_android: "95"
+  #   safari: "5"
+  #   safari_ios: "4.2"
+  - css.properties.cursor.none
```
Up to this point, these are off by one version for Edge when compared to caniuse. I'm not worried about this since it's pre-2020, but I did note it in: https://github.com/web-platform-dx/web-features/issues/1499.

ddbeck commented on features/cursor-style.yml:
```diff
@@ -0,0 +1,47 @@
+description: The `cursor` CSS property styles the pointer, allowing you to provide hints to the user on how to interact with the hovered element.
+spec: https://drafts.csswg.org/css-ui-4/#cursor
+caniuse:
+  - css3-cursors
+  - css3-cursors-newer
```
Depending on the results of https://github.com/mdn/browser-compat-data/issues/24142, we may want to include https://caniuse.com/css3-cursors-grab too.

ddbeck commented on features/cursor-style.yml:
```diff
@@ -0,0 +1,47 @@
+name: Cursor Style
```
Oh that said, we'll also need to rename the file. I think we _do_ want to use `cursor.yml`, however, giving CSS priority over, for example, SVG attributes.

ddbeck commented:
Now waiting on https://github.com/mdn/browser-compat-data/pull/24387.

jamesnw commented:
@ddbeck I updated the dist with the update BCD info, but it removed a lot of baseline support due to Safari iOS being dropped. I picked a compute_from to match Caniuse `css-cursors`, but am now wondering if I should remove the `css-cursors-newer` and `css-cursors-grab` which don't match.

ddbeck commented:
> @ddbeck I updated the dist with the updated BCD info, but it removed a lot of baseline support due to Safari iOS being dropped.

Yes, I think that's right—you shouldn't expect this all to work on all platforms. Odd but true.

> I picked a compute_from to match Caniuse `css-cursors`, but am now wondering if I should remove the `css-cursors-newer` and `css-cursors-grab` which don't match.

I think we should remove the `compute_from` completely and let it go. If you want to be especially careful, then you could add a TODO comment linking back to https://github.com/web-platform-dx/web-features/issues/1038. This feels like one of those features that you can "just" use without worrying about the implications on mobile and we ought to be able to communicate that fact.

jamesnw commented:
> > @ddbeck I updated the dist with the updated BCD info, but it removed a lot of baseline support due to Safari iOS being dropped.
> 
> Yes, I think that's right—you shouldn't expect this all to work on all platforms. Odd but true.
> 
> > I picked a compute_from to match Caniuse `css-cursors`, but am now wondering if I should remove the `css-cursors-newer` and `css-cursors-grab` which don't match.
> 
> I think we should remove the `compute_from` completely and let it go. The headline status will be correct with respect to all three caniuse features. If you want to be especially careful, then you could add a TODO comment linking back to #1038. This feels like one of those features that you can "just" use without worrying about the implications on mobile and we ought to be able to communicate that fact.

@ddbeck  Makes sense. I've removed the `compute_from` and added a TODO.

ddbeck reviewed (approved):
Thank you! 🎉
